lib/cisco_node_utils/tacacs_server_host.rb

```diff
@@ -23,41 +23,107 @@ class TacacsServerHost < NodeUtil
     attr_reader :name
     @hosts = {}
 
-    def initialize(name, create=true)
+    def initialize(name, instantiate=true, host_port=nil)
```

Comment:
It's not clear to me why you need to initialize with `host_port` now.
Is it a required part of the context now? 
If so, move `instantiate` to the end of the list since it's not part of the context.

`    def initialize(name, host_port=nil, instantiate=true)`